src/stdio.jl

```diff
@@ -52,6 +77,36 @@ function watch_stream(rd::IO, name::AbstractString)
     end
 end
 
+function send_stdio(name)
+    if verbose::Bool && !haskey(task_local_storage(), :IJulia_task)
+        task_local_storage(:IJulia_task, "send $name task")
+    end
+    send_stream(name)
+end
+
+send_stdout(t::Timer) = send_stdio("stdout")
+send_stderr(t::Timer) = send_stdio("stderr")
+
+function send_stream(name::AbstractString)
+    buf = bufs[name]
+    if buf.size > 0
+        s = takebuf_string(buf)
+        #if the last char is not valid, maybe only the start of a partially encoded UTF character
+        #was written, so we write it back to the buffer in the hope that the rest of the character will arrive soon
+        last_ind = chr2ind(s, length(s))
+        last_chrstr = s[last_ind:sizeof(s)]
+        if !isvalid(last_chrstr)
+            write(buf, last_chrstr)
+            last_ind -= 1
+        end
+        if last_ind > 0
```

Comment:
Julia certainly allows people to do `write(STDOUT, x)` for arbitrary `x`, which will often write binary data.   Therefore have to handle that case *somehow*, even if it is just to print an error message.  Jupyter requires us to send the data as a UTF8 string, so base64 seems as good as anything else.

If you could try out that code and see how it goes, that would begreat.